adapters/p2p2core/state.go

```diff
@@ -13,16 +13,19 @@ import (
 	"github.com/starknet-io/starknet-p2pspecs/p2p/proto/sync/state"
 )
 
-func AdaptStateDiff(reader core.StateReader, contractDiffs []*state.ContractDiff, classes []*class.Class) *core.StateDiff {
+func AdaptStateDiff(reader core.StateReader, contractDiffs []*state.ContractDiff, classes []*class.Class) (*core.StateDiff, error) {
 	var (
 		declaredV0Classes []*felt.Felt
 		declaredV1Classes = make(map[felt.Felt]*felt.Felt)
 	)
 
-	for _, class := range utils.Map(classes, AdaptClass) {
+	for class, err := range utils.MapWithErrors(classes, AdaptClass) {
+		if err != nil {
+			return nil, fmt.Errorf("unsupported class: %w", err)
+		}
 		h, err := class.Hash()
 		if err != nil {
-			panic(fmt.Errorf("unexpected error: %v when calculating class hash", err))
+			return nil, fmt.Errorf("unexpected error: %w when calculating class hash", err)
```

Comment:
I understand it's legacy, but let's do this instead:

```suggestion
			return nil, fmt.Errorf("unexpected error when calculating class hash: %w", err)
```